cluster-controller/src/main/java/io/strimzi/controller/cluster/ClusterController.java

```diff
@@ -201,13 +215,18 @@ private void recreateConfigMapWatch() {
     /**
       Periodical reconciliation (in case we lost some event)
      */
-    private void reconcile() {
-        kafkaClusterOperations.reconcileAll(namespace, labels);
-        kafkaConnectClusterOperations.reconcileAll(namespace, labels);
+    private Future<?> reconcile() {
+
+        List<Future> result = new ArrayList<>(2 + (kafkaConnectS2IClusterOperations != null ? 1 : 0));
```

Comment:
Do we need this kind of mathematics just to set the initial array size? (just a curious question)

Reply:
In most of the Operations we have this kind of math. The default capacity for ArrayList should be 10 if I remember well but in our case we know how many items we are adding to the list, so no wasted memory.

Reply:
If you pass a correct initial size to `ArrayList` it won't need to reallocate its internal `Object[]`. If it's too small it definitely will, if it's a little too big it doesn't matter, because it's only a few bytes wasted, but there will be no need for reallocating. So in this particular case, I guess just `new ArrayList<>(3)` would suffice.